packages/kokkos/core/src/impl/Kokkos_Core.cpp

```diff
@@ -428,6 +487,22 @@ bool check_int_arg(char const* arg, char const* expected, int* value) {
   return true;
 }
 
+bool check_string_arg(char const* arg, char const* expected, char const** value) {
+  if (!check_arg(arg, expected)) return false;
+  std::size_t arg_len = std::strlen(arg);
+  std::size_t exp_len = std::strlen(expected);
+  bool okay = true;
+  if (arg_len == exp_len || arg[exp_len] != '=') okay = false;
+  *value = arg + exp_len + 1;
```

Comment:
You should set `value` _after_ the `if (! okay)` branch, since if "not okay," then `exp_len` could be bogus.